.github/workflows/cli-release.yml

```diff
@@ -0,0 +1,21 @@
+name: cli-release
+
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build_rust:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest", "windows-latest", "macos-latest"]
```

Comment:
Are you planning on building binaries for both arm64 and amd64? (I don't feel strongly that you _have_ to, just making a note).

If so, consider that `macos-latest` runs on arm64, where `macos-13` runs on amd64. Can consider targeting both `macos-latest` and `macos-13` to get both architectures for free.

Sadly it doesn't look as simple for arm64 Linux, there doesn't appear to be runners available, so would need to either cross-compile or rent an arm64 VM from GCP. Maybe we can just skip arm64 Linux... client-side usage of arm64 Linux seems to be uncommon outside of Raspberry Pi's.

Reply:
Pertinent docs https://docs.github.com/en/actions/using-jobs/choosing-the-runner-for-a-job#standard-github-hosted-runners-for-public-repositories.

Reply:
I'm optimistic that we can cross compile from x86_64 -> aarch64 on `ubuntu-latest` and the other way around on `macos-latest`, so all we'll need is to `rustup add target <blah>` (or moral equivalent in the dtolnay toolchain action). If that doesn't work, I'll look into the `cross` action on the marketplace. That'll all be in the next PR. Right now I'm stuck getting GH to run actions at all...